src/data/quot.lean

```diff
@@ -56,6 +56,66 @@ quot.lift (quot.mk s) (λ x y rxy, quot.sound (h x y rxy))
 lemma factor_mk_eq {α : Type*} (r s : α → α → Prop) (h : ∀ x y, r x y → s x y) :
   factor r s h ∘ quot.mk _ = quot.mk _ := rfl
 
+variables {γ : Sort*} {r : α → α → Prop} {s : β → β → Prop}
+
+/-- Descends a function `f : α → β → γ` to quotients of `α` and `β`. -/
+attribute [reducible, elab_as_eliminator]
+protected def lift₂
```

Comment:
Can this be used to simplify `quotient.lift₂`?

Reply:
Yes, you could write:
```
def quotient.lift₂' {α β φ: Sort*} [s₁ : setoid α] [s₂ : setoid β] 
  (f : α → β → φ) (h : ∀ (a₁ : α) (a₂ : β) (b₁ : α) (b₂ : β),
       a₁ ≈ b₁ → a₂ ≈ b₂ → f a₁ a₂ = f b₁ b₂) :
    quotient s₁ → quotient s₂ → φ :=
quot.lift₂ f (λ a b₁ b₂ hb, h a b₁ a b₂ (setoid.refl a) hb)
             (λ a₁ a₂ b ha, h a₁ b a₂ b ha (setoid.refl b))
```
but that change would need to happen in the core library and would probably break of lot of things.

Reply:
I suppose the follow up question is, would that ever be worth it?